### PR TITLE
Rename Tendermint to Cosmos

### PR DESCRIPTION
**Pull request by pienkowb**

### CODE REVIEW

**Comment by evaporei** [comment]
Hey @pienkowb the only thing I think I have to comment on this is that instead of having to use `this` on the constructor, in AssemblyScript you can omit if you're just setting the properties.

This:

```typescript
export class HeaderOnlyBlock {
  public header: Header

  constructor(header: Header) {
    this.header = header
  }
}
```

Is the same as this 🙂 :


```typescript
export class HeaderOnlyBlock {
  // doesn't need the property here as well :)
  constructor(public header: Header) { /* no code is needed! */ }
}
```

File: package-lock.json [hunk 1]
```diff
@@ -0,0 +1,87 @@
+{
```
Comment (evaporei) [inline package-lock.json:1]: Also I don't think we want to include the `package-lock.json` file here.

Comment (pienkowb) [reply package-lock.json:1]: Sure, we'll remove this file.

**Comment by pienkowb** [comment]
> So you can probably simplify the code a bit (if you agree with the change ofc).

@evaporei I totally agree, your version looks much cleaner 👍  The issue is that we use a generator to create this file, so we would need to update the generator first. Maybe there's some kind of a linter that could do this change automatically?

**Comment by evaporei** [comment]
I don't think the linter can simplify this one unfortunately. I'll approve/merge this once the `package-lock.json` file is removed 🙂

**Comment by pienkowb** [comment]
> I don't think the linter can simplify this one unfortunately. I'll approve/merge this once the `package-lock.json` file is removed 🙂

The `package-lock.json` file is now removed. I also modified the generator to use parameter properties :)

**Review by evaporei — APPROVED** [review]

**Pull request merged**